control-plane/api/mesh/v2beta1/mesh_gateway_types.go

```diff
@@ -38,7 +38,7 @@ type MeshGateway struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
-	Spec   pbmesh.MeshGateway `json:"spec,omitempty"`
+	Spec   pbmesh.MeshGatewaySpec `json:"spec,omitempty"`
```

Comment:
So we need to also include the WorkloadSelector on this type, so that we can send it in when we register to Consul. The tricky part is making sure this is not added to the CRD, even though it will be present here.